src/array.ts

```diff
@@ -134,7 +134,7 @@ export namespace Shim {
 
 		while (count > 0) {
 			if (start in target) {
-				target[offset] = target[start];
+				(<any> target)[offset] = target[start];
```

Comment:
It appears the TypeScript team have agressively typed `ArrayLike` in TS2.0.  I would recommend that we provide a local `ArrayLike` for this, which fixes the issue here:

```typescript
interface ArrayLike<T> {
	readonly length: number;
	[offset: number]: T;
}
```

Maybe it is worth raising an issue too that this is rather aggressively typed and find out if there was some reasoning for it (I haven't looked to see if it was already raised though).